crypto/fipsmodule/rand/urandom.c

```diff
@@ -86,10 +135,29 @@ void __msan_unpoison(void *, size_t);
 #endif
 
 static ssize_t boringssl_getrandom(void *buf, size_t buf_len, unsigned flags) {
+
   ssize_t ret;
+  long backoff = INITIAL_BACKOFF_DELAY;
+  size_t retry_counter = 0;
+
   do {
     ret = syscall(__NR_getrandom, buf, buf_len, flags);
-  } while (ret == -1 && errno == EINTR);
+    if ((ret == -1) && (errno != EINTR)) {
```

Comment:
When errno == EINTR, this flow will insert a backoff, if I followed it correctly. Do we want that or we want to retry without backoff, since the original loop was retrying only in the case the call was interrupted, and was doing that without delay.

Reply:
If `errno = EINTR`, there should be no change.

Let's follow it:
* the sys call errors with `ret = -1` and `errno = EINTR`.
* We check the if conditional `(ret == -1) && (errno != EINTR)`.
* `ret = -1`, but `errno = EINTR`. Hence, condition is not satisfied.
* we drop to check the condition in the do-while loop.
* `ret = -1`, so, we retry the sys call.

This is the same behaviour as before.

Reply:
I hadn't followed it correctly, sorry. I missed `}` on l.159.